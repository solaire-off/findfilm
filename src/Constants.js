export const API_ROOT = "http://localhost:4000";
export const SITE_NAME = ["netflix", "roulette"];
export const GENRES_LIST = ["Documentary", "Comedy", "Horror", "Crime"];
export const DEFAULT_SORT_FIELD = "vote_count";
export const SORT_TYPES = [
  {
    name: "popularity",
    value: DEFAULT_SORT_FIELD,
  },
  {
    name: "release date",
    value: "release_date",
  },
  {
    name: "rating",
    value: "vote_average",
  },
  {
    name: "name",
    value: "title",
  },
];
export const FETCH_FILMS_COUNT = 18;
export const DEFAULT_GENRE_ANY = "All";
export const EXAMPLE_FILM = {
  title: "La La Land",
  tagline: "Here's to the fools who dream.",
  vote_average: 7.9,
  vote_count: 6782,
  release_date: "2016-12-29",
  poster_path:
    "https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg",
  overview:
    "Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.",
  budget: 30000000,
  revenue: 445435700,
  runtime: 128,
  genres: ["Comedy", "Drama", "Romance"],
  id: 313369,
};
export const EXAMPLE_FILM_LIST = [
  {
    id: 337167,
    title: "Fifty Shades Freed",
    tagline: "Don't miss the climax",
    vote_average: 6.1,
    vote_count: 1195,
    release_date: "2018-02-07",
    poster_path:
      "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    overview:
      "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
    budget: 55000000,
    revenue: 136906000,
    genres: ["Drama", "Romance"],
    runtime: 106,
  },
  {
    id: 269149,
    title: "Zootopia",
    tagline: "Welcome to the urban jungle.",
    vote_average: 7.7,
    vote_count: 6795,
    release_date: "2016-02-11",
    poster_path:
      "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    overview:
      "Determined to prove herself, Officer Judy Hopps, the first bunny on Zootopia's police force, jumps at the chance to crack her first case - even if it means partnering with scam-artist fox Nick Wilde to solve the mystery.",
    budget: 150000000,
    revenue: 1023784195,
    genres: ["Animation", "Adventure", "Family", "Comedy"],
    runtime: 108,
  },
  {
    id: 181808,
    title: "Star Wars: The Last Jedi",
    tagline: "The Saga Continues",
    vote_average: 7.1,
    vote_count: 4732,
    release_date: "2017-12-13",
    poster_path:
      "https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
    overview:
      "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
    budget: 200000000,
    revenue: 1325937250,
    genres: ["Fantasy", "Adventure", "Science Fiction"],
    runtime: 152,
  },
];
