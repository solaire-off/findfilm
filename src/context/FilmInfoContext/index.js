import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const FilmInfoContext = createContext();
const FilmInfoActionContext = createContext();

export const useFilmInfoContext = () => useContext(FilmInfoContext);
export const useFilmInfoActionContext = () => useContext(FilmInfoActionContext);

export const FilmInfoContextProvider = ({ children }) => {
  const [selectedFilm, setSelectedFilm] = useState(null);

  return (
    <FilmInfoContext.Provider value={selectedFilm}>
      <FilmInfoActionContext.Provider value={setSelectedFilm}>
        {children}
      </FilmInfoActionContext.Provider>
    </FilmInfoContext.Provider>
  );
};

FilmInfoContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
