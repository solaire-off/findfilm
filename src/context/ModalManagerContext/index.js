import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const ModalManagerContext = createContext();
const ModalManagerActionContext = createContext();

export const useModalManagerContext = () => useContext(ModalManagerContext);
export const useModalManagerActionContext = () =>
  useContext(ModalManagerActionContext);

export const ModalManagerProvider = ({ children }) => {
  const [activeModal, setActiveModal] = useState({
    type: null,
    props: null,
  });

  return (
    <ModalManagerContext.Provider value={activeModal}>
      <ModalManagerActionContext.Provider value={setActiveModal}>
        {children}
      </ModalManagerActionContext.Provider>
    </ModalManagerContext.Provider>
  );
};

ModalManagerProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
