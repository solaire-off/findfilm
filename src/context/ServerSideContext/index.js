import React, { createContext, useContext } from "react";
import PropTypes from "prop-types";

export const ServerSideContext = createContext(null);

export const useServerSideContext = () => useContext(ServerSideContext);

export const ServerSideProvider = ({ value, children }) => {
  return (
    <ServerSideContext.Provider value={value}>
      {children}
    </ServerSideContext.Provider>
  );
};

ServerSideProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
