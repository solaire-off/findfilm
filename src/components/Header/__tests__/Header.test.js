import React from "react";
import { render, act, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Header } from "..";
import { SITE_NAME } from "../../../Constants";
import { MemoryRouter } from "react-router-dom";
import { ModalManagerActionContext } from "../../../context/ModalManagerContext";

const mockHistoryReplace = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    replace: mockHistoryReplace,
  }),
}));

it("Current render default header", async () => {
  await act(async () => {
    render(
      <MemoryRouter initialentries={[{ pathname: "/search" }]}>
        <Header title={SITE_NAME} />
      </MemoryRouter>
    );
  });
  expect(document.body).toMatchSnapshot();
});

it("Current render header when open film info", async () => {
  await act(async () => {
    render(
      <MemoryRouter
        initialEntries={[{ hash: "", pathname: "/search", search: "?movie=1" }]}
        initialIndex={0}
      >
        <Header title={SITE_NAME} />
      </MemoryRouter>
    );
  });
  expect(document.body).toMatchSnapshot();
});

it("Work open modal", async () => {
  const mockOpenModal = jest.fn();
  await act(async () => {
    render(
      <MemoryRouter
        initialEntries={[{ hash: "", pathname: "/search" }]}
        initialIndex={0}
      >
        <ModalManagerActionContext.Provider value={mockOpenModal}>
          <Header title={SITE_NAME} />
        </ModalManagerActionContext.Provider>
      </MemoryRouter>
    );
  });
  userEvent.click(
    screen.getByRole("button", {
      name: "+ Add Movie",
    })
  );

  expect(mockOpenModal).toBeCalledTimes(1);
});

it("Change search params in history for close info about film", async () => {
  await act(async () => {
    render(
      <MemoryRouter
        initialEntries={[{ hash: "", pathname: "/search", search: "?movie=1" }]}
        initialIndex={0}
      >
        <Header title={SITE_NAME} />
      </MemoryRouter>
    );
  });

  userEvent.click(screen.getByLabelText(/close film info/i));

  expect(mockHistoryReplace).toBeCalledTimes(1);
});
