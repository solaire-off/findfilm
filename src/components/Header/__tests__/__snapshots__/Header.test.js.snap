// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render default header 1`] = `
<body>
  <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__in"
        >
          <a
            class="header__title"
            href="/search"
          >
            <span
              class="font-weight-black"
            >
              netflix
            </span>
            roulette
          </a>
          <button
            class="btn btn--outline-primary  "
            type="button"
          >
            + Add Movie
          </button>
        </div>
      </div>
    </header>
  </div>
</body>
`;

exports[`Current render header when open film info 1`] = `
<body>
  <div>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__in"
        >
          <a
            class="header__title"
            href="/search"
          >
            <span
              class="font-weight-black"
            >
              netflix
            </span>
            roulette
          </a>
          <button
            aria-label="Close film info"
            class="icon-btn"
            type="button"
          >
            <svg
              class="icon-btn__image"
              fill="none"
              height="30"
              viewBox="0 0 29 30"
              width="29"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="18.5"
                cy="10.5"
                r="9.5"
                stroke="#F65261"
                stroke-width="2"
              />
              <path
                d="M10.5 19.5L1.5 28.5"
                stroke="#F65261"
                stroke-linecap="square"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
  </div>
</body>
`;
